File: Leetcode/cpp/Bitwise AND of Numbers Range.cc
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        int u = 1;
        int ans = 0;
        while (m && n) {
            if (m == n && (m & 1)) {
                ans |= u;
            }
            m >>= 1;
            n >>= 1;
            u <<= 1;
        }
        return ans;
    }
};

File: Leetcode/python/Bitwise AND of Numbers Range.py
class Solution:
    # @param m, an integer
    # @param n, an integer
    # @return an integer
    def rangeBitwiseAnd(self, m, n):
        ans = 0
        p = 0
        while m and n:
            if (n & 1) and n == m:
                ans |= 1 << p
            p += 1
            n >>= 1
            m >>= 1
        return ans
